RecoTracker/LSTCore/src/alpaka/Triplet.h

```diff
@@ -249,20 +249,66 @@ namespace SDL {
                                                        unsigned int& firstMDIndex,
                                                        unsigned int& secondMDIndex,
                                                        unsigned int& thirdMDIndex,
-                                                       float& residual) {
+                                                       float& residual,
+                                                       float& circleRadius,
+                                                       float& circleCenterX, 
+                                                       float& circleCenterY) {
+
+    //following Philip's layer number prescription
+    const int layer1 = modulesInGPU.sdlLayers[innerInnerLowerModuleIndex];
+    const int layer2 = modulesInGPU.sdlLayers[middleLowerModuleIndex];
+    const int layer3 = modulesInGPU.sdlLayers[outerOuterLowerModuleIndex];
+
     //get the rt and z
-    const float& r1 = mdsInGPU.anchorRt[firstMDIndex];
+    const float& r1 = mdsInGPU.anchorRt[firstMDIndex]; // all the values are stored in the unit of cm, in the calculation below we need to be cautious if we want to use the meter unit
```

Comment:
still, why `&` here? the compiler may be smart enough to dereference; otherwise, it's an extra cost of keeping the address and then reading from it to the register when making a computation.

Reply:
I see, I will change it to values. Thank you!